algorithms/grandpa/prover/src/lib.rs

```diff
@@ -185,34 +185,8 @@ where
 		T::Hash: From<H::Hash>,
 		H::Number: finality_grandpa::BlockNumberOps,
 		T::BlockNumber: One,
+		F: Fn(&<T as Config>::BlockNumber) -> bool,
```

Comment:
why hooking it up through generics and not calling it inside as a normal function?